proxy-node-gateway/src/main/java/uk/gov/ida/notification/session/GatewaySessionData.java

```diff
@@ -9,25 +11,51 @@
 import javax.validation.ConstraintViolation;
 import javax.validation.Validation;
 import javax.validation.Validator;
+import javax.validation.constraints.NotNull;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
 public class GatewaySessionData {
+
+    @NotNull
     @NotEmpty
+    @JsonProperty
     private final String hubRequestId;
 
+    @NotNull
     @NotEmpty
+    @JsonProperty
     private final String eidasRequestId;
 
+    @NotNull
     @NotEmpty
+    @JsonProperty
     private final String eidasDestination;
 
+    @JsonProperty
     private final String eidasRelayState;
 
+    @NotNull
     @NotEmpty
+    @JsonProperty
     private final String eidasConnectorPublicKey;
 
+    @JsonCreator
+    public GatewaySessionData(
+        @JsonProperty("hubRequestId") String hubRequestId,
+        @JsonProperty("eidasRequestId") String eidasRequestId,
```

Comment:
Is the `@JsonProperty` annotation required on constructor args and on the field? (asking for a friend)

Reply:
Yeah it doesn't seem to work without it, I did try without. If anyone knows a better way id love to know 😃

Reply:
There is a better way! 

You just need an empty constructor and then you don't need all that `@JsonCreator` and `@JsonProperty("...")` fluff :)

See `HubResponseTranslatorRequest` for instance

```
...
    @NotNull
    @JsonProperty
    private String connectorEncryptionCertificate;

@SuppressWarnings("Needed for JSON serialisation")
    public HubResponseTranslatorRequest() {
    }

// Normal constructor without JsonCreator fluff
public HubResponseTranslatorRequest(
            String samlResponse,
            String requestId,
            String eidasRequestId,
            String levelOfAssurance,
            URI destinationUrl,
            String connectorEncryptionCertificate) {
        this.samlResponse = samlResponse;
        this.requestId = requestId;
        this.eidasRequestId = eidasRequestId;
        this.levelOfAssurance = levelOfAssurance;
        this.destinationUrl = destinationUrl;
        this.connectorEncryptionCertificate = connectorEncryptionCertificate;
    }
...
```

Reply:
😱 ❤️